src/RuleFilterParser/Api/IResult.cs

```diff
@@ -0,0 +1,11 @@
+﻿namespace RuleFilterParser;
+
+public interface IResult
```

Comment:
I think we will always force a Type. The whole idea behind this is to provide a strongly type solution. But maybe there is a use case where it might be handy...

Reply:
I agree with that is might be not necessary. This was just copied from Core API.

Actually, I hope a user would never have to interact with the `IResult` or `IResultList<>` data at all. The API result is currently only used inside the `ApiPagedEnumerable<T>` class and could be made ``internal or `private`. Maybe we could add additional Linq functions to get the metadata as well..

Reply:
interface `IResult` is now removed